src/components/Point.tsx

```diff
@@ -198,6 +200,19 @@ const Point = (props: {
 
   const imageUrl = require(`../images/${shape}.svg`);
 
+  const onClickShapeIcon = () => {
+    props.togglePoint({ pointId: point.pointId });
+
+    // TODO: Uncomment this to bring back toggling main point feature.
```

Comment:
So I remember from our discussion that you intend to have double-clicking set the main point.

In general, you do not want double-clicking to be the action that most users do. A few reasons:
- How are users going to know/remember to double-click? Maybe you could have a tutorial teaching new users how to use your app, but you should never expect users to remember shortcuts.
- Double-clicking requires some dexterity, and some users don't want to / can't physically do it consistently.

It's totally fine to have double-clicking be an option, but there should definitely be another more discoverable, physically easier way for the user to do the same thing.

In particular, I'm thinking a menu that opens up with some options that users can click. For example, maybe clicking on the icon will open up a menu with two options: "select point" and "set main point".